### PR TITLE
tidb-binlog: introduce how to upgrade binlog

### PR DESCRIPTION
**Pull request by yikeke**
### What is changed, added or deleted? 

introduce how to upgrade tidb binlog to the latest cluster version

**Note that** the first paragraph of the English version might differ a bit from the Chinese version. I did it on purpose to make it more concise.

### What is the related PR or file link(s)? 

- [ ] Reference link(s):
- [x] This PR is to align with: https://github.com/pingcap/docs-cn/pull/2175
- [ ] N/A (not applicable)

### Which TiDB version(s) does your changes apply to? 

- [x] All active versions: dev, v3.0, v2.1, v3.1
- [ ] dev (the latest development version)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] All active and inactive versions
- [ ] N/A (not applicable)

**Note:** If your changes apply to multiple TiDB versions, make sure you update the documents in the **corresponding** version folders such as "dev", "v3.0", "v2.1" and "v3.1" in this PR.

- [ ] Updated one version first. Will update other versions after I get two LGTMs.

### CODE REVIEW

File: dev/reference/tidb-binlog/upgrade.md [hunk 1]
```diff
@@ -1,11 +1,52 @@
...
+## Upgrade TiDB Binlog deployed with Ansible
+
+Follow the steps in this section if you deploy TiDB Binlog with [TiDB Ansible Playbook](https://github.com/pingcap/tidb-ansible).
+
+### Upgrade Pump
+
+First, upgrade the Pump component:
+
+1. Copy the new version of the binary `pump` file into the`({ resources_dir })/bin` directory.
+2. Execute the `ansible-playbook rolling_update.yml --tags = pump` command to perform a rolling update for Pump.
```
Comment (suzaku) [inline dev/reference/tidb-binlog/upgrade.md]:
```suggestion
2. Execute the `ansible-playbook rolling_update.yml --tags=pump` command to perform a rolling update for Pump.
```

File: v2.1/reference/tidb-binlog/upgrade.md [hunk 2]
```diff
@@ -1,11 +1,52 @@
...
+## Upgrade TiDB Binlog deployed with Ansible
+
+Follow the steps in this section if you deploy TiDB Binlog with [TiDB Ansible Playbook](https://github.com/pingcap/tidb-ansible).
+
+### Upgrade Pump
+
+First, upgrade the Pump component:
+
+1. Copy the new version of the binary `pump` file into the`({ resources_dir })/bin` directory.
+2. Execute the `ansible-playbook rolling_update.yml --tags = pump` command to perform a rolling update for Pump.
```
Comment (suzaku) [inline v2.1/reference/tidb-binlog/upgrade.md]:
```suggestion
2. Execute the `ansible-playbook rolling_update.yml --tags=pump` command to perform a rolling update for Pump.
```

File: v3.0/reference/tidb-binlog/upgrade.md [hunk 3]
```diff
@@ -1,11 +1,52 @@
...
+## Upgrade TiDB Binlog deployed with Ansible
+
+Follow the steps in this section if you deploy TiDB Binlog with [TiDB Ansible Playbook](https://github.com/pingcap/tidb-ansible).
+
+### Upgrade Pump
+
+First, upgrade the Pump component:
+
+1. Copy the new version of the binary `pump` file into the`({ resources_dir })/bin` directory.
+2. Execute the `ansible-playbook rolling_update.yml --tags = pump` command to perform a rolling update for Pump.
```
Comment (suzaku) [inline v3.0/reference/tidb-binlog/upgrade.md]:
```suggestion
2. Execute the `ansible-playbook rolling_update.yml --tags=pump` command to perform a rolling update for Pump.
```

File: v3.1/reference/tidb-binlog/upgrade.md [hunk 4]
```diff
@@ -1,10 +1,51 @@
...
+## Upgrade TiDB Binlog deployed with Ansible
+
+Follow the steps in this section if you deploy TiDB Binlog with [TiDB Ansible Playbook](https://github.com/pingcap/tidb-ansible).
+
+### Upgrade Pump
+
+First, upgrade the Pump component:
+
+1. Copy the new version of the binary `pump` file into the`({ resources_dir })/bin` directory.
+2. Execute the `ansible-playbook rolling_update.yml --tags = pump` command to perform a rolling update for Pump.
```
Comment (suzaku) [inline v3.1/reference/tidb-binlog/upgrade.md]:
```suggestion
2. Execute the `ansible-playbook rolling_update.yml --tags=pump` command to perform a rolling update for Pump.
```

**Comment by yikeke** [comment]
PTAL @lilin90

**Pull request merged**